### PR TITLE
CICD: Fix the release note generation.

### PR DESCRIPTION
**Pull request by mikeller**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [x] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Escape special characters in the input to the string substitution.

### CODE REVIEW

**Review by dirkhh — COMMENTED** [review]
oh. my.
I'm sure there is a better way to do this. Mine clearly was stupid.

File: scripts/create-releasenotes.sh [hunk 1]
```diff
@@ -4,5 +4,9 @@
 #
 # Usage: create-releasenotes.sh pr_num pr_url pr_title commit_id commit_url
 
-sed "s/PRNUM/$1/;s/PRURL/$2/;s/PRTITLE/$3/;s/COMMITID/$4/;s/COMMITURL/$5/" < gh_release_notes.in > gh_release_notes
+prurl=$(sed -e 's/[&\\/]/\\&/g; s/$/\\/' -e '$s/\\$//' <<< "$2")
+prtitle=$(sed -e 's/[&\\/]/\\&/g; s/$/\\/' -e '$s/\\$//' <<< "$3")
+commiturl=$(sed -e 's/[&\\/]/\\&/g; s/$/\\/' -e '$s/\\$//' <<< "$5")
```
Comment (dirkhh) [inline scripts/create-releasenotes.sh]: I've been doing some googling, and there are many other things that we'd have to quote. sed is a bit of a mess.
awk would be much easier... using `sub(pattern, input)` you only need to escape `\` and`&`...

**Comment by mikeller** [comment]
> oh. my. I'm sure there is a better way to do this. Mine clearly was stupid.

XSLT? /ducks

Back to seriousness, I think the 'keep it as simple as possible' approach will be fine for this.

**Comment by dirkhh** [comment]
> > oh. my. I'm sure there is a better way to do this. Mine clearly was stupid.
> 
> XSLT? /ducks
> 
> Back to seriousness, I think the 'keep it as simple as possible' approach will be fine for this.

Oh, I agree. I just realize that the response could contain any URL characters, so a few more that sed would stumble over.
But I'm happy to see us try what you have, TBH.

**Review by dirkhh — COMMENTED** [review]
don't hate me... your commit message no longer matches...

**Comment by dirkhh** [comment]
OK, I think I have a better way to do this. Much better, actually.

Can you take a look at this action run: https://github.com/dirkhh/testrepo/actions/runs/7467660626/job/20321592864

Click on the "now from a script" step

Here I just print the created release notes to stdout.

As I was playing with this, I actually thought that for more complex merge requests pointing to the last commit wasn't that useful and to drop that part and just link to the pull request used.

What do you think, should I redo the releasenote creation here based on this?

**Comment by dirkhh** [comment]
Actually, @mikeller -- this one is better: https://github.com/dirkhh/testrepo/actions/runs/7467726058/job/20321766525
The other one still had odd line breaks included.

**Comment by dirkhh** [comment]
ditto, also should be closed, right?

**Pull request closed without merging**